tests/functional/unit_testing/test_unit_testing.py

```diff
@@ -271,6 +273,42 @@ def test_no_this_input(self, project):
 """
 
 
+class TestUnitTestIncrementalModelWithAlias:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_incremental_model.sql": my_incremental_model_with_alias_sql,
+            "events.sql": event_sql,
+            "schema.yml": test_my_model_incremental_yml_basic,
+        }
+
+    def test_basic(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 2
+
+        # Select by model name
+        results = run_dbt(["test", "--select", "my_incremental_model"], expect_pass=True)
+        assert len(results) == 2
```

Comment:
This tests the regression reported in https://github.com/dbt-labs/dbt-core/issues/10754.

I confirmed this test fails without the fix but succeeds with the fix in place:
```shell
python -m pytest tests/functional/unit_testing/test_unit_testing.py::TestUnitTestIncrementalModelWithAlias
```